migration/transitioning.rst

```diff
@@ -0,0 +1,92 @@
+.. _transitioning_mt:
+
+Migrating an existing single-node PostgreSQL database to Citus sometimes requires adjusting the schema and queries for optimal performance. Citus extends PostgreSQL with distributed functionality, but it is not a drop-in replacement that scales out all workloads. A performant Citus cluster involves thinking about the data model, tooling, and choice of SQL features used.
+
+Migration tactics differ between the two main Citus use cases of multi-tenant applications and real-time analytics. The former requires fewer data model changes so we'll begin there.
+
+Multi-tenant Data Model
+=======================
+
+Citus is well suited to hosting B2B multi-tenant application data. In this model application tenants share a Citus cluster and a schema. Each tenant's table data is stored in a shard determined by a configurable tenant id column. Citus pushes queries down to run directly on the relevant tenant shard in the cluster, spreading out the computation. Once queries are routed this way they can be executed without concern for the rest of the cluster. These queries can use the full features of SQL, including joins and transactions, without running into the inherent limitations of a distributed system.
+
+This section will explore how to model for the multi-tenant scenario, including necessary adjustments to the schema and queries.
+
+The general process is
+
+1. Recreate the schema of the original database on the Citus coordinator node.
+2. Modify the schema for multi-tenant distribution.
+3. Copy the data from the original database to Citus.
+
+Schema Migration
+----------------
+
+Transitioning from a standalone database instance to a sharded multi-tenant system requires identifying and modifying three types of tables which we may term *per-tenant*, *reference*, and *global*. The distinction hinges on whether the tables have (or reference) a column serving as tenant id. The concept of tenant id depends on the application and who exactly are considered its tenants.
+
+Consider an example multi-tenant application similar to Etsy or Shopify where each tenant is a store. Here's a portion of a simplified schema:
+
+.. image:: ../images/erd/mt-before.png
```

Comment:
The tool I'm using ([erd](https://github.com/BurntSushi/erd)) manages the layout. Internally it's calling graphviz which does have the ability to honor graph layout preferences. In the future we could open an issue in erd to allow layout specification or I could use graphviz's `dot` tool directly. Or another relation diagram tool if you know a good one.

Also I tried appending " (pk)" and " (fk)" to lines but the program couldn't parse those characters. I'll add a legend to the schema diagrams that explains the font conventions.